src/rgw/rgw_gc.cc

```diff
@@ -38,6 +39,21 @@ void RGWGC::initialize(CephContext *_cct, RGWRados *_store) {
     snprintf(buf, 32, ".%d", i);
     obj_names[i].append(buf);
   }
+
+  // Create GC queues
+  for (int i = 0; i < max_objs; i++) {
+    ldpp_dout(this, 20) << "RGWGC::initialize initing gc queue with name = " << obj_names[i] << dendl;
+    librados::ObjectWriteOperation op;
+    op.assert_exists();
+    uint64_t queue_size = 1048576, num_urgent_data_entries = 50;
+    cls_rgw_gc_init_queue(op, obj_names[i], queue_size, num_urgent_data_entries);
+    int ret = store->gc_operate(obj_names[i], &op);
+    if (ret == -ENOENT) {
+      ldpp_dout(this, 20) << "RGWGC::initialize creating gc queue with name = " << obj_names[i] << dendl;
+      cls_rgw_gc_create_queue(op, obj_names[i], queue_size, num_urgent_data_entries);
+      store->gc_operate(obj_names[i], &op);
+    }
+  }
```

Comment:
@cbodley @mattbenjamin : I have added a new init queue method that basically initializes the head if its not already initalized, (to work around the issue of create queue being called 6 times on the osd side and rgw getting back a file exists errror). Here the rgw logs show that init queue is being invoked and the osd reply return value is 0. When I check the osd logs, I can see that create queue is being invoked once successfully for each queue, and then init queue is invoked 5 times, returning 0 without the head being initialized since it has already been initialized by create queue. This solves the issue of the queue getting created and the head being initialized properly.
1. But I don't know why init queue is being invoked more than once for each queue. 
2. Also I don't know from where create queue is being invoked, since rgw logs show an osd op for init queue for each queue.

Reply:
Also, I think we cant re-use the existing gc object names, as that might be a problem in the upgrade scenarios. Will modify that bit.

Reply:
depending on how we implement the upgrade/migration, reusing the existing objects could potentially make things easier; i.e. we may want a cls call that reads entries out of omap and writes them into the queue